File: sem4/MPP/proiect/appClientServer/CompetitionPersistence/src/main/java/org/example/persistence/repository/ChildDBRepository.java
package org.example.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.example.model.Child;
import org.example.persistence.IChildRepository;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ChildDBRepository implements IChildRepository {
    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public ChildDBRepository(Properties props) {
        logger.info("Initializing ChidDBRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }
    @Override
    public void add(Child elem) {
        logger.traceEntry("saving task {} ", elem);
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into child (child_name, child_age) values (?, ?)")){
            preStmt.setString(1, elem.getName());
            preStmt.setInt(2, elem.getAge());
            int result= preStmt.executeUpdate();
            logger.trace("Saved {} instances", elem);
        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Child elem) {

    }

    @Override
    public void update(Child elem, Integer id) {

    }

    @Override
    public Child findById(Integer id) {
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preStmt=con.prepareStatement("select * from child where id_child = ?")){
            preStmt.setInt(1, id);
            try (ResultSet result= preStmt.executeQuery()){
                while (result.next()){
                    int id_child = result.getInt("id_child");
                    String name = result.getString("child_name");
                    int age = result.getInt("child_age");
                    Child child = new Child(name,age);
                    child.setId(id_child);
                    return child;
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public Iterable<Child> findAll() {
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        List<Child> children = new ArrayList<>();
        try (PreparedStatement preStmt=con.prepareStatement("select * from child")){
            try (ResultSet result= preStmt.executeQuery()){
                while (result.next()){
                    int id_child = result.getInt("id_child");
                    String name = result.getString("child_name");
                    int age = result.getInt("child_age");
                    Child child = new Child(name,age);
                    child.setId(id_child);
                    children.add(child);
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit(children);
        return children;
    }
}

File: sem4/MPP/proiectC#/AppClientServer/Persistence/ChildDbRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using log4net;
using model;
namespace repository;

public class ChildDbRepository : IChildRepository
{
    private static readonly ILog log = LogManager.GetLogger("SortingTaskDbRepository");

    public ChildDbRepository()
		{
			log.Info("Creating SortingTaskDbRepository ");
			
		}

		public Child FindOne(int id)
		{
			log.InfoFormat("Entering findOne with value {0}", id);
			IDbConnection con = DBUtils.getConnection();

			using (var comm = con.CreateCommand())
			{
				comm.CommandText = "select * from child where id_child=@id";
				IDbDataParameter paramId = comm.CreateParameter();
				paramId.ParameterName = "@id";
				paramId.Value = id;
				comm.Parameters.Add(paramId);

				using (var dataR = comm.ExecuteReader())
				{
					if (dataR.Read())
					{
						int idC = dataR.GetInt32(0);
						String name = dataR.GetString(1);
						int age = dataR.GetInt32(2);
						Child child = new Child(name, age);
						child.ID = idC;
						log.InfoFormat("Exiting findOne with value {0}", child);
						return child;
					}
				}
			}
			log.InfoFormat("Exiting findOne with value {0}", null);
			return null;
		}

		public List<Child> FindAll()
		{
			IDbConnection con = DBUtils.getConnection();
			List<Child> children = new List<Child>();
			using (var comm = con.CreateCommand())
			{
				comm.CommandText = "select * from child";

				using (var dataR = comm.ExecuteReader())
				{
					while (dataR.Read())
					{
						int idC = dataR.GetInt32(0);
						String name = dataR.GetString(1);
						int age = dataR.GetInt32(2);
						Child child = new Child(name, age);
						child.ID = idC;
						children.Add(child);
					}
				}
			}
			return children;
		}
		
		public Child Save(Child entity)
		{
			var con = DBUtils.getConnection();

			using (var comm = con.CreateCommand())
			{
				comm.CommandText = "insert into child(child_name, child_age) values (@name, @age)";

				var paramName = comm.CreateParameter();
				paramName.ParameterName = "@name";
				paramName.Value = entity.name;
				comm.Parameters.Add(paramName);

				IDbDataParameter paramAge = comm.CreateParameter();
				paramAge.ParameterName = "@age";
				paramAge.Value = entity.age.ToString();
				comm.Parameters.Add(paramAge);

				var result = comm.ExecuteNonQuery();
				if (result == 0)
					//throw new RepositoryException("No task added !");
					throw new Exception("No task added !");
				else
				{
					return entity;
				}
			}
			
		}

		public Child Delete(int id)
		{
			return null;
		}

		public Child Update(Child entity)
		{
			return null;
		}
}